pkg/registry/policy/rest/storage_policy.go

```diff
@@ -35,22 +35,27 @@ func (p RESTStorageProvider) NewRESTStorage(apiResourceConfigSource serverstorag
 	// If you add a version here, be sure to add an entry in `k8s.io/kubernetes/cmd/kube-apiserver/app/aggregator.go with specific priorities.
 	// TODO refactor the plumbing to provide the information in the APIGroupInfo
 
-	if apiResourceConfigSource.VersionEnabled(policyapiv1beta1.SchemeGroupVersion) {
+	if apiResourceConfigSource.AnyResourcesForVersionEnabled(policyapiv1beta1.SchemeGroupVersion) {
 		apiGroupInfo.VersionedResourcesStorageMap[policyapiv1beta1.SchemeGroupVersion.Version] = p.v1beta1Storage(apiResourceConfigSource, restOptionsGetter)
 		apiGroupInfo.GroupMeta.GroupVersion = policyapiv1beta1.SchemeGroupVersion
 	}
 	return apiGroupInfo, true
 }
 
 func (p RESTStorageProvider) v1beta1Storage(apiResourceConfigSource serverstorage.APIResourceConfigSource, restOptionsGetter generic.RESTOptionsGetter) map[string]rest.Storage {
+	version := policyapiv1beta1.SchemeGroupVersion
 	storage := map[string]rest.Storage{}
-	// poddisruptionbudgets
-	poddisruptionbudgetStorage, poddisruptionbudgetStatusStorage := poddisruptionbudgetstore.NewREST(restOptionsGetter)
-	storage["poddisruptionbudgets"] = poddisruptionbudgetStorage
-	storage["poddisruptionbudgets/status"] = poddisruptionbudgetStatusStorage
 
-	storage["podsecuritypolicies"] = pspstore.NewREST(restOptionsGetter)
+	// poddisruptionbudgets
+	if apiResourceConfigSource.ResourceEnabled(version.WithResource("poddisruptionbudgets")) {
+		poddisruptionbudgetStorage, poddisruptionbudgetStatusStorage := poddisruptionbudgetstore.NewREST(restOptionsGetter)
+		storage["poddisruptionbudgets"] = poddisruptionbudgetStorage
+		storage["poddisruptionbudgets/status"] = poddisruptionbudgetStatusStorage
+	}
 
+	if apiResourceConfigSource.ResourceEnabled(version.WithResource("podsecuritypolicies")) {
```

Comment:
Should here also support `podsecuritypolicy` as extentions group?

Reply:
> Should here also support podsecuritypolicy as extentions group?

No.  separate makes sense